plugin/pkg/admission/priority/admission.go

```diff
@@ -196,8 +216,12 @@ func (p *priorityPlugin) admitPod(a admission.Attributes) error {
 			pod.Spec.PriorityClassName = pcName
 		} else {
 			pcName := pod.Spec.PriorityClassName
-			if !priorityClassPermittedInNamespace(pcName, a.GetNamespace()) {
-				return admission.NewForbidden(a, fmt.Errorf("pods with %v priorityClass is not permitted in %v namespace", pcName, a.GetNamespace()))
+			// If ResourceQuotaScopeSelectors is enabled, we should let pods with critical priorityClass to be created
+			// any namespace where administrator wants it to be created.
+			if !p.featureGates.resourceQuotaFeatureGateEnabled {
```

Comment:
This is a risky change. In our current code, we protect clusters from users who may create pods with system-critical priority causing preemption of important workloads. The protection is enabled regardless of existence of quota or whether the quota feature is enabled. With this change, users can create pods with system-critical priority if quota is disabled.

Reply:
You mean users can create pods with system-critical priority in any namespace when resourceQuotaScopeSelectors is disabled? If yes, I don't think so - You cannot create critical pods in any namespace other than `kube-system`, if you try to create cluster-critical pods, it would throw an error.

Reply:
You are right. I missed it initially.